modules/service/src/main/scala/lucuma/odb/graphql/mapping/ObservationMapping.scala

```diff
@@ -98,36 +103,51 @@ trait ObservationMapping[F[_]]
               OrderBy(OrderSelections(List(OrderSelection[ObsAttachment.Id](ObsAttachmentType / "id"))), child)
             )
           )
+
+        case Select("itc", List(
+          BooleanBinding.Option("useCache", rUseCache)
+        ), child) =>
+          rUseCache.map { useCache =>
+            Environment(
+              Env(UseCacheParam -> useCache.getOrElse(UseCacheDefault)),
+              Select("itc", Nil, child)
+            )
+          }
       }
     )
 
   def itcQueryHandler: EffectHandler[F] =
     new EffectHandler[F] {
-      private def extractIds(queries: List[(Query, Cursor)]): Result[List[(Program.Id, Observation.Id)]] =
+      private def extractIds(queries: List[(Query, Cursor)]): Result[List[(Program.Id, Observation.Id, Boolean)]] =
         queries.traverse { case (_, cursor) =>
           for {
             p <- cursor.fieldAs[Program.Id]("programId")
             o <- cursor.fieldAs[Observation.Id]("id")
-          } yield (p, o)
+            c <- cursor.fullEnv.getR[Boolean](UseCacheParam)
+          } yield (p, o, c)
         }
 
-      private def callItc(pid: Program.Id, oid: Observation.Id): F[Result[(Observation.Id, Itc.ResultSet)]] =
+      private def callItc(
+        pid:      Program.Id,
+        oid:      Observation.Id,
+        useCache: Boolean
+      ): F[Result[((Observation.Id, Boolean), Itc.ResultSet)]] =
         services.useTransactionally {
           itc(itcClient)
-            .lookup(pid, oid, useCache = true)
+            .lookup(pid, oid, useCache)
             .map {
               case Left(errors)     => Result.failure(errors.map(_.format).intercalate(", "))
-              case Right(resultSet) => (oid, resultSet).success
+              case Right(resultSet) => ((oid, useCache), resultSet).success
             }
         }
 
       def runEffects(queries: List[(Query, Cursor)]): F[Result[List[(Query, Cursor)]]] =
```

Comment:
To be honest, I'm not sure under what circumstances the queries would be "batched" and when "batched" whether each cursor's environment will be the same.  I'm assuming possibly not, so the results for two such queries with the same observation id might disagree about whether to use the cache.  I wish this could be simplified though.

Reply:
As I understand it the environments will differ in general, and no batching or anything is done for you. But the handler can do whatever it wants (including batching).